x-pack/plugins/security_solution/server/lib/detection_engine/signals/__mocks__/es_results.ts

```diff
@@ -910,6 +910,23 @@ export const sampleDocSearchResultsNoSortIdNoHits = (
   },
 });
 
+export const repeatedHitsWithSortId = (
+  count: number,
+  guids: string[],
+  ips?: Array<string | string[]>,
+  destIps?: Array<string | string[]>,
+  sortIds?: string[]
+): SignalSourceHit[] => {
+  return Array.from({ length: count }).map((x, index) => ({
+    ...sampleDocWithSortId(
+      guids[index],
+      sortIds,
+      ips ? ips[index] : '127.0.0.1',
+      destIps ? destIps[index] : '127.0.0.1'
```

Comment:
Regarding the previous comment, maybe we could have `127.0.0.1` be the default Ip and destIp for any index that doesn't exist? You could default the values to `[]` if not provided and do something like `destIps[index] ?? '127.0.0.1'` instead. Same for `ips`.